src/image/Filter.cc

```diff
@@ -168,6 +173,64 @@ std::vector<std::string> Filter::getNames() {
     return names;
 }
 
+std::shared_ptr<typehandling::Storable> Filter::cloneStorable() const {
+    return std::make_unique<Filter>(*this);
+}
+
+bool Filter::equals(typehandling::Storable const& other) const noexcept {
+    return singleClassEquals(this, other);
+}
+
+namespace {
+
+struct PersistenceHelper {
+    table::Schema schema;
+    table::Key<std::string> name;
+
+    static PersistenceHelper const& get() {
+        static PersistenceHelper const instance;
+        return instance;
+    }
+
+private:
+    PersistenceHelper() : schema(), name(schema.addField<std::string>("name", "name of the filter")) {
+        schema.getCitizen().markPersistent();
+    }
+};
+
+class FilterFactory : public table::io::PersistableFactory {
+public:
+    std::shared_ptr<table::io::Persistable> read(InputArchive const& archive,
+                                                 CatalogVector const& catalogs) const override {
+        PersistenceHelper const& keys = PersistenceHelper::get();
+        LSST_ARCHIVE_ASSERT(catalogs.size() == 1u);
+        LSST_ARCHIVE_ASSERT(catalogs.front().getSchema() == keys.schema);
+        return std::make_shared<Filter>(catalogs.front().begin()->get(keys.name));
+    }
```

Comment:
Hmm, must have misunderstood how the registry was populated.

Do you recommend removing `Storable` entirely, or keeping the other `Storable` capabilities and making `isPersistable` return `false`?

Reply:
On second thought, are you sure there's a problem with `Filter` depersistence? It looks like the existing code simply [calls the constructor with force=true](https://github.com/lsst/afw/blob/master/src/image/ExposureFitsReader.cc#L85).

Reply:
I think you're right - for some reason I had it in my head that it was `force=false` right now, and moreover that it couldn't easily be changed.  No idea what my past self was thinking.

I think that means we should keep this implementation, but when we get to reimplementing `Exposure`, we should pause and think about whether to use the new `Filter` persistence there, just because that's another on-disk format change we'll have to support going forward.  I don't think that's a problem if we continue to special-case `Filter` by also writing the name into the header.

Reply:
Honestly, I'm not sure any generic persistence framework will be backwards-compatible with the existing file format, even for "normal" components. For DM-17982 I'm considering leaving the persistence code as-is, and supporting unknown `Storable`s later.